.github/workflows/release-layer-collector.yml

```diff
@@ -31,6 +40,11 @@ jobs:
         with:
           name: opentelemetry-collector-layer-${{ matrix.architecture }}.zip
           path: ${{ github.workspace }}/collector/build/opentelemetry-collector-layer-${{ matrix.architecture }}.zip
+      - name: Add Binary to Release
+        run: |
+          gh release upload ${{github.ref_name}} ${{ github.workspace }}/collector/build/opentelemetry-collector-layer-${{ matrix.architecture }}.zip
```

Comment:
Thanks for contributing this.

This job is triggered by tag creation. The release is usually created afterwards. So if the release isn't created yet, does this create a published release or in draft mode?

Reply:
So originally it would publish the release. I have updated the PR to create a draft release and upload the binary to the draft. Example drafts are here: https://github.com/webfinesse/opentelemetry-lambda/releases

Here is a screenshot since I am not sure if drafts are publicly seen or not:

<img width="1332" alt="image" src="https://github.com/user-attachments/assets/e3e0c453-40ac-41a6-8de3-000cba5e0a7c">

Reply:
Awesome.  If you can customize the draft title to match our current pattern, then great.  Otherwise I think I'm good with this. Thanks!

Reply:
I went ahead and updated this. 
<img width="1091" alt="image" src="https://github.com/user-attachments/assets/d0ddecd8-b153-4bbf-acb4-e0db65e79e99">